### PR TITLE
Improve handling of unscalable mods and fix some mods from alternate qualities not applying.

### LINKED ISSUE #4848: Vaal Clarity and potnetially other auras set stats scale with aura effect.
### Check version

- [X] I'm running the latest version of Path of Building and I've verified this by checking the [changelog](https://github.com/PathOfBuildingCommunity/PathOfBuilding/blob/master/CHANGELOG.md)

### Check for duplicates

- [X] I've checked for duplicate issues by using the search function of the [issue tracker](https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues)

### What is the expected behaviour/value?

"Skills cost no mana" should always make skills cost no mana indpendent of aura effect of vaal clarity.

### What is the actual behaviour/value?

Vaal Clarities "Skills cost no mana" sets 100% less mana cost. This can be scaled with aura effect to no longer make skills cost no mana.

### How to reproduce the issue

1. Add Vaal Clarity
2. Add #% reduced aura effect to custom mods
3. Add a skill with a mana cost.
4. See that it has a non-zero mana cost.

### Character build code

```shell
https://pobb.in/0DvHmPGNBnWQ
```

### Screenshots

image

### PR DESCRIPTION
**Pull request by Paliak**
Fixes #4848 #4849 #5031 #5042

Absorbed [#5043](https://github.com/PathOfBuildingCommunity/PathOfBuilding/pull/5043).

This pr changes the way unscaled mods are handled and fixes related issues adding unscaled tag to a variety of mods. Also fixes some mods from alternate qualities not applying correctly and implements parsing for a few currently unhandled stats.

### Before screenshot:
obraz

### After screenshot:
obraz

### CODE REVIEW

**Comment by Paliak** [comment]
Breakdown for impurities currently broken. Working on a fix.

**Comment by Nostrademous** [comment]
@Paliak why couldn't `0, 0, { type = "GlobalEffect", effectType = "Aura", unscalable = true }` part be added at the `SkillStatMap.lua` level to all the mods?

**Comment by Paliak** [comment]
I looked at how purity of elements worked and applied that to the impurities.

File: src/Data/Skills/act_dex.lua [hunk 1]
```diff
@@ -312,6 +312,12 @@ skills["ArcticArmour"] = {
 		["new_arctic_armour_fire_damage_taken_when_hit_+%_final"] = {
 			mod("FireDamageTakenWhenHit", "MORE", nil, 0, 0, { type = "Condition", var = "Stationary" }, { type = "GlobalEffect", effectType = "Buff" }),
 		},
+		["base_immune_to_freeze"] = {
+			--Display only
+		},
+	},
+	baseMods = {
+		mod("AvoidFreeze", "BASE", 100, 0, 0, { type = "GlobalEffect", effectType = "Aura", unscalable = true }),
```
[the remark this answers is not in the document]
Comment (Paliak) [reply src/Data/Skills/act_dex.lua]: Fixed.

**Comment by QuickStick123** [comment]
Fixes https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues/5031

**Review by LocalIdentity — APPROVED** [review]

**Pull request merged**